dds/idl/dds_generator.h

```diff
@@ -568,30 +590,113 @@ std::string getEnumLabel(AST_Expression* label_val, AST_Type* disc)
   return e.replace(colon + 2, std::string::npos, label);
 }
 
+template <typename IntType>
+std::ostream& signed_int_helper(std::ostream& o, IntType value, IntType min)
+{
+  /*
+   * It seems that in C/C++ the minus sign and the bare number are parsed
+   * separately for negative integer literals. This can cause compilers
+   * to complain when using the minimum value of a signed integer because
+   * the number without the minus sign is 1 past the max signed value.
+   *
+   * https://stackoverflow.com/questions/65007935
+   *
+   * Apparently the workaround is to write it as `VALUE_PLUS_ONE - 1`.
+   */
+  const bool min_value = value == min;
+  if (min_value) ++value;
+  o << value;
+  if (min_value) o << " - 1";
+  return o;
+}
+
+inline
+std::ostream& hex_value(std::ostream& o, unsigned value, size_t bytes)
+{
+  std::ios saved(0);
+  saved.copyfmt(o);
```

Comment:
Use `RestoreOutputStreamState` helper?

Reply:
Yes I can do that, but [`copyfmt`](https://en.cppreference.com/w/cpp/io/basic_ios/copyfmt) seems to be the way the library provides for saving and restoring stream state, so I will change `RestoreOutputStreamState` to use it.